File: JavaScript/test/PayrollApplication.test.js
import Employee from '../src/Employee.mjs'
import PayCheck from '../src/PayCheck.mjs'
import PayrollApplication from '../src/PayrollApplication.mjs'

describe('PayrollApplication', () => {
  const IRRELEVANT = 53

  it('Without bonus', () => {
    const app = new PayrollApplication()
    const employee = new Employee(100, false, false)

    const payCheck = app.payAmount(employee, 30)

    expect(payCheck.equals(new PayCheck(3000, "EMP"))).toBe(true)
  })

  it('With bonus', () => {
    const app = new PayrollApplication()
    const employee = new Employee(10, false, false)

    const payCheck = app.payAmount(employee, 41)

    expect(payCheck.equals(new PayCheck(1410, "EMP"))).toBe(true)
  })

  it('Retired', () => {
    const app = new PayrollApplication()
    const employee = new Employee(IRRELEVANT, false, true)

    const payCheck = app.payAmount(employee, IRRELEVANT)

    expect(payCheck.equals(new PayCheck(0, "RET"))).toBe(true)
  })

  it('Separated', () => {
    const app = new PayrollApplication()
    const employee = new Employee(IRRELEVANT, true, false)

    const payCheck = app.payAmount(employee, IRRELEVANT)

    expect(payCheck.equals(new PayCheck(0, "SEP"))).toBe(true)
  })

  it('Separated and Retired', () => {
    const app = new PayrollApplication()
    const employee = new Employee(IRRELEVANT, true, true)

    const payCheck = app.payAmount(employee, IRRELEVANT)

    expect(payCheck.equals(new PayCheck(0, "SEP"))).toBe(true)
  })
})

File: Python/tests/payroll_application_test.py
from payroll_application import Employee, PayCheck, pay_amount

IRRELEVANT = 53


def test_without_bonus():
    employee = Employee(100, False, False)
    pay_check = pay_amount(employee, 30)
    assert pay_check == PayCheck(3000, "EMP")


def test_with_bonus():
    employee = Employee(10, False, False)
    pay_check = pay_amount(employee, 41)
    assert pay_check == PayCheck(1410, "EMP")


def test_retired():
    employee = Employee(IRRELEVANT, False, True)
    pay_check = pay_amount(employee, IRRELEVANT)
    assert pay_check == PayCheck(0, "RET")


def test_separated():
    employee = Employee(IRRELEVANT, True, False)
    pay_check = pay_amount(employee, IRRELEVANT)
    assert pay_check == PayCheck(0, "SEP")


def test_separated_and_retired():
    employee = Employee(IRRELEVANT, True, True)
    pay_check = pay_amount(employee, IRRELEVANT)
    assert pay_check == PayCheck(0, "SEP")
